server/gitlab.go

```diff
@@ -180,18 +222,16 @@ func hasSameEnvs(info *E2ETestTriggerInfo, glVars []*gitlab.PipelineVariable) (b
 					return false, nil
 				}
 			}
-			if glVar.Key == envKeyBuildTag && glVar.Value != info.BuildTag {
-				return false, nil
-			}
-
 			if k == glVar.Key && v == glVar.Value {
 				i++
+				optsCount := len(info.EnvVars)
+				if optsCount == i {
+					return true, nil
```

Comment:
If the `glVar` array contains all the options before the `PR_NUMBER`, then this line would run, even though the code has never checked if the PR_NUMBER is the same.
Not sure if the code could allow such a case right now, but it's very easy to look over

Reply:
Ah I now notice this only happens inf `len(info.EnvVars)` is 1, in any other case the PR_NUMBER check is done at least once in that first loop

Reply:
Added the failing test case, and updated the logic to fix that